src/core/inspect/check.rs

```diff
@@ -0,0 +1,54 @@
+use std::path::Path;
+
+use anyhow::{bail, Result};
+use log::warn;
+
+use super::{inspect, kernel_version::KernelVersionReq};
+
+/// Check various prerequisites for Retis to properly work, allowing to bail out
+/// early and to explain what isn't compatible to the user. This helper can also
+/// be used to show warning or information that do not prevent Retis from
+/// starting.
+pub(crate) fn collection_prerequisites() -> Result<()> {
```

Comment:
Something that would be absolutely awesome would be to verify the process has CAP_EBPF! Not necessarily for this PR

Reply:
as a note, at least access to debugfs and kallsyms (kptr_restrict = 0, perf_event_paranoid = 0) is needed.
Not sure there's something else missing. I don't think it's a goal for the time being, maybe for the future.